src/main/java/entity_gateway/CalendarManager.java

```diff
@@ -0,0 +1,13 @@
+package main.java.entity_gateway;
+
+import main.java.use_case.CalendarEventModel;
+
+import java.util.List;
+
+public interface CalendarManager {
+
+    boolean addEvent(CalendarEventModel eventData);
+
+    List<EventReader> getAllEvents();
+
```

Comment:
```
    boolean addEvent(CalendarEventModel eventData);

    List<EventReader> getAllEvents();
```
Why do we need EventReader and CalendarEventModel? I know that the former is returned when we are getting Event from the database and the latter is used when we are adding new Events (from UI). But why can't we combine them into one such that the UI and DB both interact with the same model interface?

Reply:
So this is similar to the topic on whether you should use entities as boundary request/response models. The answer is "No" because we should consider these models as coincidentally similar to the entity (someone should be able to write the boundary without knowledge of the entities). Although the actual content is the same, the responsibilities are not: if we end up wanting to add a new implicitly calculated getter in `EventReader`, do we want `CalendarEventModel` to update, too? And if we want to split the combined class back into two, which and how many references do we have to manually update? Things like this make your software hard and unchangeable.